### ISSUE TITLE
Parsing derived types with multiple namespaces doesn't work for single-package style models

### ISSUE THREAD

**Issue by untereiner** [tefra/xsdata#520]
Hello,

I have an issue when I parse an xml into the classes previously generated.
My error is:
`xsdata.exceptions.ParserError: Unknown property {http://www.energistics.org/energyml/data/resqmlv2}AbstractIntegerArray:{http://www.energistics.org/energyml/data/commonv2}NullValue
`

That is a simplified version of the schema:
```xml
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
  <xs:complexType name="AbstractIntegerArray" abstract="true">    
  </xs:complexType>

  <xs:complexType name="IntegerExternalArray">
    <xs:extension base="AbstractIntegerArray">
    </xs:extension>
    <xs:element name="NullValue" type="xs:long">
    </xs:element>
  </xs:complexType>

  <xs:complexType name="TriangulatedSetRepresentation">
    <xs:element name="SupportingRepresentationNodes" type="AbstractIntegerArray" minOccurs="1" maxOccurs="1">                            
    </xs:element>
  </xs:complexType>
</xs:schema>

```

And here is a sample xml file
```xml
<?xml version="1.0" encoding="UTF-8"?>
<TriangulatedSetRepresentation>
		<SupportingRepresentationNodes xsi:type="IntegerExternalArray">
                <NullValue xsi:type="xsd:long">2147483647</NullValue>
        </SupportingRepresentationNodes>
</TriangulatedSetRepresentation>
```

I think the parser expects an `AbstractIntegerArray` and do not know what to do with the `NullValue`.
I am using master branch

**Comment by untereiner** [comment]
Here is the complete example just in case.
Usage: `python ./test.py TriangulatedSetRepresentation_17004506-f1db-4d34-8d66-efd6b4b15d73.xml`
[complete_example.zip](https://github.com/tefra/xsdata/files/6639766/complete_example.zip)

**Comment by tefra** [comment]
The single package structure style doesn't work very well for derived types (aka xsi:type="eml23:IntegerExternalArray") when more than one namespaces is involved.  [docs](https://xsdata.readthedocs.io/en/latest/codegen.html#circular-imports)

If you notice on top of each module xsdata adds the source namespace the classes/types were originally declared. 
Unfortunately the single package structure automatically bundles all types under the same namespace, which is extracted from the first class of the module.  

I suggest to switch to another structure style, if you picked the single package due to circular imports give the new `clusters` style a try.

Suggestions:
1. Generate the source namespace as a meta property when there is more than one namespace involved per module
2. Raise an exception during codegen for this case if the above doesn't work.

**Comment by untereiner** [comment]
The `clusters` style is working fine thank you.
I am maybe wrong but I am trying to round trip between xml and json.
I can deserialize an xml file and serialize it into a json string.
But when I try to deserialize this json string (to  serialize it to xml) I have an issue similar to the mentioned above:

```xml
xsdata/xsdata/formats/dataclass/parsers/json.py", line 131, in bind_dataclass
    raise ParserError(f"Unknown property {clazz.__qualname__}.{key}")
xsdata.exceptions.ParserError: Unknown property AbstractIntegerArray.NullValue
```

**Comment by tefra** [comment]
Roundtrips between json and xml also don't work all the time with derived/abstract types, you see the information `xsi:type` is lost between transfers.

We could make the parser attempt all possible classes, but that will also have some misses, in case more than one subclass fits the json properties.

**Comment by untereiner** [comment]
One another option could be to serialize into badgerFish json

**Comment by tefra** [comment]
> One another option could be to serialize into badgerFish json

Will continue this one here #523 


Original issue was resolved in #525, the generator now will check for multiple namespaces in the module and will generate the target namespace per type instead on top for all module classes
